src/Inkluzitron/Modules/ImagesModule.cs

```diff
@@ -24,15 +27,82 @@ namespace Inkluzitron.Modules
     public class ImagesModule : ModuleBase
     {
         private ProfilePictureService ProfilePictureService { get; }
+        private BotDatabaseContext DbContext { get; set; }
+        private DatabaseFactory DatabaseFactory { get; }
 
-        public ImagesModule(ProfilePictureService profilePictureService)
+
+        public ImagesModule(ProfilePictureService profilePictureService, DatabaseFactory databaseFactory)
         {
+            DatabaseFactory = databaseFactory;
             ProfilePictureService = profilePictureService;
 
             if (!Directory.Exists("ImageCache"))
                 Directory.CreateDirectory("ImageCache");
         }
 
+        protected override void BeforeExecute(CommandInfo command)
+        {
+            DbContext = DatabaseFactory.Create();
+            base.BeforeExecute(command);
+        }
+
+        protected override void AfterExecute(CommandInfo command)
+        {
+            DbContext?.Dispose();
+            base.AfterExecute(command);
+        }
+
+        private static readonly double WhipBdsmTreshold = 0.5; // TODO config maybe?
+
+        /// <summary>
+        /// Validation based on BDSM results.
+        /// This check prevents sub users from whipping dom users
+        /// </summary>
+        /// <param name="source">Initiator (whipping user)</param>
+        /// <param name="target">Target (user to be whipped)</param>
+        /// <returns>If source user can whip target user</returns>
+        private async Task<bool> CanWhipUser(IUser source, IUser target)
```

Comment:
Raději bych tohle měl vytažené jako samostatnou službu, která by tuhle funkci nabízela ostatním. Aby ses v ImagesModule nemusel starat o to, jak jsou ty testy vnitřně uložené v DB.

Reply:
Možná raději službu, která by nabízela něco jako .IsDom(IUser), .IsSub(IUser), .IsSwitch(IUser)

Reply:
Enterprise go brrrr

Reply:
Zatím mi to přijde zbytečný

Reply:
V 02d6f60 jsem udělal službu na získávání vlastností uživatele z posledního testu. Zatím to implementuje jen metody, které jsou zde používány, ale není problém v budoucnu jednoduše rozšířit